cloudinit/unknown.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudinit
+
+import (
+	"encoding/json"
+
+	"github.com/pkg/errors"
+	"sigs.k8s.io/kind/pkg/exec"
+)
+
+type unknown struct {
+	module string
+	lines  []string
+}
+
+func newUnknown(module string) action {
+	return &unknown{module: module}
+}
+
+// Unmarshal will unmarshall unknown actions and slurp the value
+func (u *unknown) Unmarshal(data []byte) error {
+	// try unmarshalling to a slice of strings
+	var s1 []string
+	if err := json.Unmarshal(data, &s1); err != nil {
+		if _, ok := err.(*json.UnmarshalTypeError); !ok {
+			return errors.WithStack(err)
+		}
+	} else {
+		u.lines = s1
+		return nil
+	}
+
+	// If it's not a slice of strings it should be one string value
+	var s2 string
```

Comment:
I don't think we can know that it will only be a []string or string, since we are subject to all the yaml types.

For both of the module additions I'm working on they also have nested lists for example

Reply:
Yeah...I noticed that this existing module would break when we paired this morning.

Is it ok to get this merged in and revisit the base assumption that everything will be either a string or []string?